packages/transformers/webextension/src/WebExtensionTransformer.js

```diff
@@ -279,18 +361,24 @@ export default (new Transformer({
       '@parcel/transformer-webextension',
       'Invalid Web Extension manifest',
     );
+    asset.setEnvironment({
```

Comment:
What's the reason for this? Generally we should avoid it as it can cause issues. If it is needed, you might also need to move it to the top of the transformer so that it is set before adding any dependencies. Setting the environment overrides the asset's id, which is also used in those dependencies, so  setting it later means the dependencies might not point to the right asset id anymore.

Reply:
This is before `collectDependencies`, but I can move it to the top for clarity maybe. I know this is a bad practice but for web extensions it is literally impossible to use it in any environment but a browser, and this makes it way easier for people to use the web extension config without modification. The `inline` and `inlineSources` source mapping options especially are critical for source mapping to work by default.